docs/tools/editor/vscode.md

```diff
@@ -1,79 +1,129 @@
 
 ![](./images/vscode-1.jpg)
 
-安装完成后，打开需要运行的文件，点击右上角的小三角图标即可运行代码；按下快捷键<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>N</kbd>（在 macOS 下是<kbd>Control</kbd>+<kbd>Option</kbd>+<kbd>N</kbd>）也可以得到同样的效果。
+After finishing installation, open file that need to be run, click the upper-right little triangle icon to run code. Pressing shortcut key <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>N</kbd>(or, for macOS, <kbd>Control</kbd>+<kbd>Option</kbd>+<kbd>N</kbd>) will do the same
 
 ???+ warning
-    如果安装了 VS Code 与 Code Runner 后，代码仍然无法运行，很有可能是因为系统尚未安装 C/C++ 的运行环境，参考 [Hello, World! 页面](../../lang/helloworld.md) 以安装。
-    
-    记得勾选设置中的 Run In Terminal 选项，如图：![](./images/vscode-7.png)
+    If code is still unable to run after install VS Code and Code Runner, it it probably because of the lack of C/C++ running environment in the operating system. For installation of environment please refer to Hello, World!](../../lang/helloworld.md).
+
+    Remember to toggle on Run In Terminal in preferences as the following image: ![](./images/vscode-7.png)
 
-## 使用 C/C++ 插件编译并调试
+## Compile and Debug using C/C++ extensions
 
-### 安装插件
+### Installation
 
-在 VS Code 中打开插件商店，在搜索栏中输入 `C++` 或者 `@category:"programming languages"`，然后找到 C/C++，点击 Install 安装插件。
+Open extension store in VS Code, type `C++` or `@category:"programming languages"`, and find C/C++, and press Install to install it.
 
 ![](./images/vscode-2.png)
 
 ???+ warning
-    在配置前，请确保系统已经安装了 G++ 或 Clang，并已添加到了 `PATH` 中。请使用 CMD 或者 PowerShell，而不是 Git Bash 作为集成终端。
+    Before continue to config, please make sure the that system has installed a C++ or Clang, and has added them to `$PATH`. Please use CMD or Powershell instead of Git Bash.
 
-### 配置编译
+### Configuring Compilation
 
-首先用 VS Code 打开一个文件夹，然后按下<kbd>F1</kbd>，输入 `C/C++: Edit configurations (UI)`，进入 C/C++ 插件的设置界面。
+First, open a folder in VS Code, and then press <kbd>F1</kbd>, type `C/C++: Edit configurations (UI)` and enter the configuration page of C/C++ extension.
 
 ![vscode-3](images/vscode-3.png)
 
-在“编译器路径”中选择 G++ 或 Clang 的所在路径。如果没有可选项，请检查编译器所在路径是否添加到了操作系统的 `PATH` 变量中。
+Choose the path where G++ or Clang located in Compiler Path. re-check whether the path of compiler has been added to `PATH` environment variable of operating system if there is nothing to choose.
 
-### 配置 IntelliSense
+### Configuring IntelliSense
 
-用于调整 VS Code 的智能补全。
+This section is talking about adjusting smart auto-completion of VS Code.
 
-如果你使用 Clang 编译器，在“IntelliSense 模式”中选择 `clang-x64` 而非默认的 `msvc-x64`；如果你使用 G++ 编译器，选择 `gcc-x64` 以使用自动补全等功能。否则会得到“IntelliSense 模式 msvc-x64 与编译器路径不兼容。”的错误。
+If you are using Clang as compiler, remember to choose `clang-x64` instead of default `msvc-x64`, or for G++ compiler users, choose `gcc-x64`, to enable features like auto-completion. Otherwise you will encounter an error "Intellisense mode msvc-x64 is incompatible with compiler path".
```

Comment:
```suggestion
If you are using Clang or G++ as compiler, remember to choose `clang-x64` (`gcc-x64` for G++) instead of default `msvc-x64` to enable features like auto-completion. Otherwise you will encounter an error saying "Intellisense mode msvc-x64 is incompatible with compiler path".
```